src/argilla/client/feedback.py

```diff
@@ -38,6 +39,7 @@
 
 if TYPE_CHECKING:
     import httpx
+    from datasets import Dataset
```

Comment:
Do we need this lib as an explicit dep in pyproject.toml?

Reply:
Hmm at least for the `ArgillaTrainer` dependencies we're assuming everything is optional, so I can either add a `try-except ImportError` block there to enable the methods just if the required dependencies are available, add `datasets` as a core dependency, or just add the `requires_version` decorator to check the version before calling any of those methods.